File: domain/src/testFixtures/java/com/study/platform/support/fake/FakeNotificationRepository.java
package com.study.platform.support.fake;

import com.study.platform.domain.notification.model.Notification;
import com.study.platform.domain.notification.model.NotificationRepository;
import org.springframework.test.util.ReflectionTestUtils;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;

public class FakeNotificationRepository implements NotificationRepository {

    private final Map<UUID, Notification> store = new HashMap<>();

    @Override
    public Notification save(Notification notification) {
        if (notification.getId() == null) {
            ReflectionTestUtils.setField(notification, "id", UUID.randomUUID());
        }
        store.put(notification.getId(), notification);
        return notification;
    }

    @Override
    public Optional<Notification> findById(UUID id) {
        return Optional.ofNullable(store.get(id));
    }

    @Override
    public void markAllAsRead(UUID receiverId) {
        store.values().stream()
                .filter(n -> n.getReceiver().getId().equals(receiverId))
                .forEach(Notification::markAsRead);
    }

    public long countByReceiverIdAndIsReadFalse(UUID receiverId) {
        return store.values().stream()
                .filter(n -> n.getReceiver().getId().equals(receiverId) && !n.isRead())
                .count();
    }

    public java.util.List<Notification> findAllByReceiverId(UUID receiverId) {
        return store.values().stream()
                .filter(n -> n.getReceiver().getId().equals(receiverId))
                .toList();
    }
}

File: domain/src/testFixtures/kotlin/com/study/platform/support/fake/FakeNotificationRepository.kt
package com.study.platform.support.fake

import com.study.platform.domain.notification.model.Notification
import com.study.platform.domain.notification.model.NotificationRepository
import java.util.UUID

class FakeNotificationRepository : AbstractFakeUuidRepository<Notification>(), NotificationRepository {

    override fun idOf(entity: Notification): UUID? = entity.id
    override fun hasTimestamps() = true

    override fun save(notification: Notification): Notification = saveEntity(notification)

    override fun markAllAsRead(receiverId: UUID) {
        store.values
            .filter { n -> n.receiver?.id == receiverId }
            .forEach { it.markAsRead() }
    }

    fun countByReceiverIdAndIsReadFalse(receiverId: UUID): Long =
        store.values.count { n -> n.receiver?.id == receiverId && !n.isRead }.toLong()

    fun findAllByReceiverId(receiverId: UUID): List<Notification> =
        store.values.filter { n -> n.receiver?.id == receiverId }
}
